src/main/java/org/sqlite/jdbc3/JDBC3DatabaseMetaData.java

```diff
@@ -1078,7 +1078,10 @@ public ResultSet getColumns(String c, String s, String tblNamePattern, String co
                             colType = colType.substring(0, iStartOfDimension).trim();
                         }
 
-                        int colGenerated = "2".equals(colHidden) ? 1 : 0;
+                        int colGenerated =
+                                Arrays.binarySearch(new String[] {"2", "3"}, colHidden) >= 0
```

Comment:
it seems convoluted instead of doing a OR condition

Reply:
Modified. @gotson Please take another look.